### ISSUE TITLE
Regression in `logprobs` functionality since v0.2.6

### ISSUE THREAD

**Issue by yourbuddyconner** [abetlen/llama-cpp-python#774]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest code. Development is very rapid so there are no tagged versions as of now.
- [x] I carefully followed the [README.md](https://github.com/abetlen/llama-cpp-python/blob/main/README.md).
- [x] I [searched using keywords relevant to my issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests) to make sure that I am creating a new issue that is not already open (or closed).
- [x] I reviewed the [Discussions](https://github.com/abetlen/llama-cpp-python/discussions), and have a new bug or useful enhancement to share.

# Expected Behavior

API requests using the llama-cpp-python[server] package with `logprobs=1` to return logprobs. 

I have this working once downgrading to `v0.2.6` like so: 

```
{
    "id": "cmpl-0186d6e8-14ce-47cd-a124-b9c7b311e83f",
    "object": "text_completion",
    "created": 1696100698,
    "model": "foo",
    "choices": [
        {
            "text": " PEDESTRIAN: TRUE",
            "index": 0,
            "logprobs": {
                "text_offset": [
                    1901,
                    1903,
                    1904,
                    1906,
                    1908,
                    1910,
                    1912,
                    1913
                ],
                "token_logprobs": [
                    -0.06649367795467859,
                    -0.0029681826316075898,
                    -2.9360146636669766e-05,
                    -2.7622959892180096e-05,
                    -2.0081132315312498e-05,
                    -1.0935793964347871e-05,
                    -0.4439232848706048,
                    -0.043034566308009586
                ],
                "tokens": [
                    " P",
                    "E",
                    "DE",
                    "ST",
                    "RI",
                    "AN",
                    ":",
                    " TRUE"
                ],
                "top_logprobs": [
                    {
                        " P": -0.06649367795467859
                    },
                    {
                        "E": -0.0029681826316075898
                    },
                    {
                        "DE": -2.9360146636669766e-05
                    },
                    {
                        "ST": -2.7622959892180096e-05
                    },
                    {
                        "RI": -2.0081132315312498e-05
                    },
                    {
                        "AN": -1.0935793964347871e-05
                    },
                    {
                        ":": -0.4439232848706048
                    },
                    {
                        " TRUE": -0.043034566308009586
                    }
                ]
            },
            "finish_reason": "stop"
        }
    ],
    "usage": {
        "prompt_tokens": 504,
        "completion_tokens": 8,
        "total_tokens": 512
    }
}
```

# Current Behavior

In the current v0.2.10 package, this currently returns the following error: 

```
openai.error.APIError: 'llama_model_params' object has no attribute 'logits_all' {"error":{"message":"'llama_model_params' object has no attribute 'logits_all'","type":"internal_server_error","param":null,"code":null}} 500 {'error': {'message': "'llama_model_params' object has no attribute 'logits_all'", 'type': 'internal_server_error', 'param': None, 'code': None}} {'date': 'Sat, 30 Sep 2023 18:58:09 GMT', 'server': 'uvicorn', 'content-length': '137', 'content-type': 'application/json', 'x-request-id': '62599f43956a4763b6d6e1b84d10ee2b'}
```

# Environment and Context

Please provide detailed information about your computer setup. This is important in case the issue is not reproducible except for under certain specific conditions.

* Physical (or virtual) hardware you are using, e.g. for Linux:

```
Architecture:                    x86_64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
Address sizes:                   48 bits physical, 48 bits virtual
CPU(s):                          64
On-line CPU(s) list:             0-63
Thread(s) per core:              2
Core(s) per socket:              32
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       AuthenticAMD
CPU family:                      25
Model:                           8
Model name:                      AMD Ryzen Threadripper PRO 5975WX 32-Cores
Stepping:                        2
Frequency boost:                 enabled
CPU MHz:                         1800.000
CPU max MHz:                     7006.6401
CPU min MHz:                     1800.0000
BogoMIPS:                        7186.41
Virtualization:                  AMD-V
L1d cache:                       1 MiB
L1i cache:                       1 MiB
L2 cache:                        16 MiB
L3 cache:                        128 MiB
NUMA node0 CPU(s):               0-63
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; Retpolines, IBPB conditional, IBRS_FW, STIBP always-on, RSB filling
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mm
                                 xext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmu
                                 lqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy sv
                                 m extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb b
                                 pext perfctr_llc mwaitx cpb cat_l3 cdp_l3 invpcid_single hw_pstate ssbd mba ibrs ibpb stibp vmmcall fsgsbase bmi1 
                                 avx2 smep bmi2 erms invpcid cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_ll
                                 c cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd amd_ppin arat npt lbrv svm_loc
                                 k nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spe
                                 c_ctrl umip pku ospke vaes vpclmulqdq rdpid overflow_recov succor smca fsrm
```

* Operating System, e.g. for Linux:

```
Linux bigthink 5.15.0-46-generic #49~20.04.1-Ubuntu SMP Thu Aug 4 19:15:44 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

* SDK version, e.g. for Linux:

```
$ python3 --version
Python 3.11.5

$ make --version
GNU Make 4.2.1
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

$ g++ --version
g++ (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

# Failure Information (for bugs)

The only notable thing I did here was compile the library with HipBLAS enabled for support with AMD GPUs. 

This currently doesnt work:
```
CC=/opt/rocm/llvm/bin/clang CXX=/opt/rocm/llvm/bin/clang++ CMAKE_ARGS="-DLLAMA_HIPBLAS=on" pip install --upgrade --force-reinstall  --no-cache-dir llama-cpp-python[server]

python -m llama_cpp.server --model models/7B/<model>.gguf --n_gpu_layers 35  --host 0.0.0.0
```

This currently works: 
```
CC=/opt/rocm/llvm/bin/clang CXX=/opt/rocm/llvm/bin/clang++ CMAKE_ARGS="-DLLAMA_HIPBLAS=on" pip install --upgrade --force-reinstall  --no-cache-dir llama-cpp-python[server]==0.2.6

python -m llama_cpp.server --model models/7B/<model>.gguf --n_gpu_layers 35  --host 0.0.0.0
```

Judging by the releases recently, there's been some churn in the library itself, so I suspect those changes have caused a regression in the code path that generates logprobs. 

I don't currently think that my specific system's configuration is causing a mess here, but happy to provide additional debugging info upon request.

**Comment by abetlen** [comment]
@yourbuddyconner yes the issue is really that this wasn't caught by type checking because both `self.context` and `self.model` are opaque `void *` so it was missed by static analysis and the ci tests don't really catch issues like this because they mock the llama.cpp library.

I'll fix this and add some better local tests using real models to the roadmap.

Cheers.

**Comment by abetlen** [comment]
@yourbuddyconner should be fixed in v0.2.11

**Comment by yourbuddyconner** [comment]
Tested, works! You dropped this, King -> 👑
